app/buck2_forkserver/src/run.rs

```diff
@@ -353,7 +354,20 @@ pub fn maybe_absolutize_exe<'a>(
 
     let abs = spawned_process_cwd.join(exe);
     if fs_util::try_exists(&abs).context("Error absolute-izing executable")? {
-        return Ok(abs.into_path_buf().into());
+        let metadata = fs::metadata(&abs).context("Error getting metadata for path")?;
```

Comment:
This looks like a bit of a race condition to me. Can you add `fs_util::metadata_if_exists` instead and then use that?

Reply:
Good catch! Done.